pex/interpreter.py

```diff
@@ -188,19 +187,9 @@ def get(cls, binary=None):
         internal_entries = frozenset(
             (pythonpath.split(os.pathsep) if pythonpath else []) + list(third_party.exposed())
         )
-
-        def is_internal_entry(entry):
-            # type: (str) -> bool
-            if entry in internal_entries:
-                return True
-            if not os.path.isabs(entry):
-                return False
-            for internal_entry in internal_entries:
-                if internal_entry == commonpath((internal_entry, entry)):
-                    return True
-            return False
-
-        sys_path = OrderedSet(entry for entry in sys.path if entry and not is_internal_entry(entry))
+        sys_path = OrderedSet(
```

Comment:
This is the simpler check that existed pre-#2328. It was always the right check; it's just that `third_party.exposed()` was broken at PEX runtime.

Reply:
And all is well here. Performing this change 1st led to a failing `test_boot_identification_leak` test, and then fixing below got that passing again.